src/snowflake/connector/file_transfer_agent.py

```diff
@@ -208,7 +213,7 @@ def __call__(self, bytes_amount: int):
         with self._lock:
             if self._output_stream:
                 self._seen_so_far += bytes_amount
-                percentage = float(self._seen_so_far / self._size)
+                percentage = percent(self._seen_so_far, self._size)
```

Comment:
`self._size` is initialized as `float(filesize)` so you are not respecting the type hints of `percent`